stl/inc/filesystem

```diff
@@ -1663,6 +1663,18 @@ namespace filesystem {
             return _Result;
         }
 
+        for (const auto& _File_name : relative_path()) {
+            if (!_Parse_root_name(_File_name.native()).empty()) {
+                return _Result;
+            }
+        }
+
+        for (const auto& _File_name : _Base.relative_path()) {
+            if (!_Parse_root_name(_File_name.native()).empty()) {
+                return _Result;
+            }
+        }
+

```

Comment:
These loops are reasonable, but would it be clearer to extract them into a function returning `bool`, named something like `_Relative_path_contains_root_name`? That would avoid code duplication, and extending the test on lines 1661-1662 would mirror the Standardese. I'm fine with the code as-is, though.

Reply:
what about using `std::any_of` in the lines 1661-1662 ?

Reply:
That's a good question (yay STL algorithms!). You could use `any_of` to implement a named function. Attempting to use it directly in lines 1661-1662 would be messy because (1) you want to iterate through `relative_path()` and `_Base.relative_path()` but `any_of` wants separate `begin` and `end` iterators, and (2) you'd need to repeat a fairly lengthy lambda (or name the lambda, in which case why not just extract the whole thing?).